crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let aria_roles = ctx.aria_roles();
+        let element_name = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+        let attributes = extract_attrs(&node.attributes());
+        let elm = element_name.text_trimmed();
+
+        if let Some(attributes_ref) = attributes.as_ref() {
+            if !is_interactive_handler_present(attributes_ref) {
+                return None;
+            }
+        } else {
+            return None;
+        }
+
+        let default_attributes = FxHashMap::default();
+        let attributes_ref = attributes.as_ref().unwrap_or(&default_attributes);
+
+        let attributes_option = if !attributes_ref.is_empty() {
+            Some(attributes_ref.clone())
+        } else {
+            None
+        };
+
+        if has_truthy_ariahidden_attr(node) {
+            return None;
+        }
+
+        let is_valid_element = match elm {
+            "section" => ["aria-label", "aria-labelledby"].iter().any(|&attr_name| {
+                node.find_attribute_by_name(attr_name)
+                    .map_or(false, |attr| {
+                        attr.as_static_value()
+                            .map_or(false, |val| !val.text().is_empty())
+                    })
+            }),
+            "a" => node.find_attribute_by_name("href").map_or(false, |attr| {
+                attr.as_static_value()
+                    .map_or(false, |val| !val.text().is_empty())
+            }),
+            _ => {
+                (!aria_roles.is_not_interactive_element(
+                    element_name.text_trimmed(),
+                    attributes_option.clone(),
+                ) && !is_invalid_element(elm))
+                    || is_valid_element(elm)
+            }
+        };
+
+        if is_valid_element {
+            return None;
+        }
+
+        if attributes_option.is_some() {
+            match node.find_attribute_by_name("role") {
+                Some(attr) => {
+                    let role_value = attr.as_static_value()?;
+                    let role_text = role_value.text();
+
+                    if has_valid_role(role_text) || aria_roles.is_role_interactive(role_text) {
+                        return None;
+                    }
+                }
+                None => {
+                    return Some(());
+                }
+            };
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            node.range(),
+            markup! {{"Avoid non-native interactive elements. If using native HTML is not possible, add an appropriate role and support for tabbing, mouse, keyboard, and touch inputs to an interactive content element"}},
+        ))
```

Comment:
you should use `RuleDiagnostic::note` method to provide detail information.

Reply:
@unvalley 
fixed in this commit 494a4aa7da01cf4d102721309c6fc1745348a933

Reply:
I left a suggestion
https://github.com/biomejs/biome/pull/2981/files#r1624767756

Reply:
@unvalley 
fixed in this https://github.com/biomejs/biome/pull/2981/commits/53745f3d518a3ef7739c01afc67bd7a126e37432